flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/utils/JsonUtils.java

```diff
@@ -0,0 +1,228 @@
+		public static final String INDEX = "index";
+		public static final String INTERVAL = "interval";
+		public static final String ENABLED = "enabled";
+		public static final String TIMEOUT = "timeout";
+		public static final String MIN_PAUSE = "min_pause";
+		public static final String MAX_CONCURRENT = "max_concurrent";
+
+		private Keys() {
+		}
+	}
+
+	public static void writeJobDetailOverviewAsJson(AccessExecutionGraph graph, JsonGenerator gen) throws IOException {
+		CurrentJobsOverviewHandler.writeJobDetailOverviewAsJson(WebMonitorUtils.createDetailsForJob(graph), gen, System.currentTimeMillis());
+	}
+
+	public static void writeMinMaxAvg(JsonGenerator gen, MinMaxAvgStats minMaxAvg) throws IOException {
+		gen.writeNumberField(Keys.MIN, minMaxAvg.getMinimum());
+		gen.writeNumberField(Keys.MAX, minMaxAvg.getMaximum());
+		gen.writeNumberField(Keys.AVG, minMaxAvg.getAverage());
+	}
+
+	public static void addIOMetrics(MutableIOMetrics summedMetrics, ExecutionState state, @Nullable IOMetrics ioMetrics, @Nullable MetricFetcher fetcher, String jobID, String taskID, int subtaskIndex) {
+		if (state.isTerminal()) {
+			if (ioMetrics != null) { // execAttempt is already finished, use final metrics stored in ExecutionGraph
+				summedMetrics.addNumBytesInLocal(ioMetrics.getNumBytesInLocal());
+				summedMetrics.addNumBytesInRemote(ioMetrics.getNumBytesInRemote());
+				summedMetrics.addNumBytesOut(ioMetrics.getNumBytesOut());
+				summedMetrics.addNumRecordsIn(ioMetrics.getNumRecordsIn());
+				summedMetrics.addNumRecordsOut(ioMetrics.getNumRecordsOut());
+			}
+		} else { // execAttempt is still running, use MetricQueryService instead
+			if (fetcher != null) {
+				fetcher.update();
+				MetricStore.SubtaskMetricStore metrics = fetcher.getMetricStore().getSubtaskMetricStore(jobID, taskID, subtaskIndex);
+				if (metrics != null) {
+					summedMetrics.addNumBytesInLocal(Long.valueOf(metrics.getMetric(MetricNames.IO_NUM_BYTES_IN_LOCAL, "0")));
+					summedMetrics.addNumBytesInRemote(Long.valueOf(metrics.getMetric(MetricNames.IO_NUM_BYTES_IN_REMOTE, "0")));
+					summedMetrics.addNumBytesOut(Long.valueOf(metrics.getMetric(MetricNames.IO_NUM_BYTES_OUT, "0")));
+					summedMetrics.addNumRecordsIn(Long.valueOf(metrics.getMetric(MetricNames.IO_NUM_RECORDS_IN, "0")));
+					summedMetrics.addNumRecordsOut(Long.valueOf(metrics.getMetric(MetricNames.IO_NUM_RECORDS_OUT, "0")));
+				}
+			}
+		}
+	}
+
+	public static void writeIOMetrics(JsonGenerator gen, IOMetrics metrics) throws IOException {
+		gen.writeObjectFieldStart(Keys.METRICS);
+		gen.writeNumberField(Keys.READ_BYTES, metrics.getNumBytesInLocal() + metrics.getNumBytesInRemote());
+		gen.writeNumberField(Keys.WRITE_BYTES, metrics.getNumBytesOut());
+		gen.writeNumberField(Keys.READ_RECORDS, metrics.getNumRecordsIn());
+		gen.writeNumberField(Keys.WRITE_RECORDS, metrics.getNumRecordsOut());
+		gen.writeEndObject();
+	}
+
+	public static class MutableIOMetrics extends IOMetrics {
```

Comment:
I think the name is fine; having it in runtime-web already implies that it is only used by the web frontend.